onnxruntime/python/session.py

```diff
@@ -72,6 +72,21 @@ def set_providers(self, providers):
         self._reset_session()
         self._load_model(providers)
 
+    def set_provider_options(self, provider, provider_options):
+
+        if not provider_options or not isinstance(provider_options, dict):
+            raise ValueError("Please specify provider options in python dict format")
+
+        if provider not in self._providers:
+            raise ValueError("{} does not in the list of registered providers {}".format(
+                provider, self._providers))
+
+        for key, val in provider_options.items():
+            provider_options[key] = str(val)
+
+        self._sess.set_device_options(provider, provider_options)
+        self._load_model(self._providers)
```

Comment:
if someone has already called set_providers() the session is created.
you can't call load_model() again without calling _reset_session()
this is the same as set_providers()

Reply:
maybe this is not the most efficient way. since you would have to reset and recreate the session